uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -323,12 +318,70 @@ define([
       }
 
       /**
+       * Initialize open mode of the dialog
+       *
+       * @return {Promise}
+       */
+      function initOpenMode() {
+        if (vm.leaveRequest.id) {
+          vm.mode = 'edit';
+          //todo in future
+          //if owner and status is approved then view only mode
+          // if(vm.role === 'owner' && vm.leaveRequest.status_id == valueOfRequestStatus('approved')){
+          //   vm.mode = 'view';
+          // }
+        } else {
+          vm.mode = 'create';
+        }
+        return $q.resolve(vm.mode);
+      }
+
+      /**
+       * Initialize user's role
+       *
+       * @return {Promise}
+       */
+      function initUserRole() {
+        if (directiveOptions.leaveRequest &&
```

Comment:
Instead of checking `ç` in many functions, we can check it in the first line of the controller and return if `false`. Or wrap to whole controller logic inside
`if(directiveOptions.leaveRequest){ ....controller....}`